### ISSUE TITLE
TypeError: Cannot read property 'name'/'type' of undefined

### ISSUE THREAD

**Issue by gaearon** [babel/babel-eslint#51]
I tried upgrading to ESLint 0.17 and latest babel-eslint and got a load of crashes, almost in each file where I use ES6 module syntax.

First one: 

```
/usr/local/lib/node_modules/eslint/lib/rules/no-func-assign.js:42
                if (def.name.name === name && def.type === "FunctionName") {
                            ^
TypeError: Cannot read property 'name' of undefined
    at checkIfIdentifierIsFunction (/usr/local/lib/node_modules/eslint/lib/rules/no-func-assign.js:42:29)
    at checkIfIdentifierIsFunction (/usr/local/lib/node_modules/eslint/lib/rules/no-func-assign.js:56:20)
```

on

```js
import shallowEqual from 'react/lib/shallowEqual';

export default function(spec) {
  return {
    getInitialState() {
      this.subscribe((key, value) => {
        data[key] = value;
      });
    }
  }
};
```

Next one:

```
/usr/local/lib/node_modules/eslint/lib/rules/no-unused-vars.js:43
            if (definition.type === "VariableDeclarator") {
                          ^
TypeError: Cannot read property 'type' of undefined
    at isExported (/usr/local/lib/node_modules/eslint/lib/rules/no-unused-vars.js:43:27)
    at EventEmitter.Program:exit (/usr/local/lib/node_modules/eslint/lib/rules/no-unused-vars.js:120:70)
    at emitOne (events.js:82:20)
    at EventEmitter.emit (events.js:166:7)
```

on

```js
import shallowEqual from 'react/lib/shallowEqual';

export default function(spec) {
  return {
    getInitialState() {
    }
  };
};
```

**Comment by btmills** [comment]
These are being addressed in eslint/eslint#2076 and eslint/eslint#2061.

**Comment by gaearon** [comment]
This is fixed with ESLint 0.17.1.
